### ISSUE TITLE
Clicking + to add annotation scrolls page back up

### ISSUE THREAD

**Issue by macobo** [PostHog/posthog#4570]
## Bug description

https://user-images.githubusercontent.com/148820/120482356-b819d980-c3b9-11eb-97b2-02451d08fc61.mp4

This makes adding annotations especially annoying

## Expected behavior

No scrolling

## How to reproduce

1. Go to insights
2. Try to add insight
3. See page scroll up

## Environment

- [x] PostHog Cloud
- [x] self-hosted PostHog, version/commit: latest master

**Comment by samwinslow** [comment]
Curious to see how this gets solved... I noticed this behavior affects a few text inputs as well.

**Comment by macobo** [comment]
Probably a `stopPropagation` for the link click event? :)

**Comment by mariusandra** [comment]
Did a quick test and removing `autoFocus` from the textarea fixes it... but that's not an ideal solution. I'll try upgrading antd next.

**Comment by mariusandra** [comment]
Nope, no change from updating antd. Not sure what's causing the issue then.

**Comment by alexkim205** [comment]
> Curious to see how this gets solved... I noticed this behavior affects a few text inputs as well.

@samwinslow Hmm interesting, I'm gonna take a stab at this one now, but could you point me to specific text inputs where we see similar issues? Maybe I can batch all the fixes up in one PR if they're related

**Comment by alexkim205** [comment]
Okay after a bit of digging I think I see what's happening.

## Some Context

There are two types of `AnnotationMarker`'s we generate for our graphs: tooltips for [existing annotations](https://github.com/alexkim205/posthog/blob/master/frontend/src/scenes/insights/LineGraph/LineGraph.js#L404-L424) and [modals for creating an annotation](https://github.com/alexkim205/posthog/blob/master/frontend/src/scenes/insights/LineGraph/LineGraph.js#L427-L455). For each marker, we create a full-height, full-width, absolutely positioned div and insert the modal as a child.

```
// For each annotation marker modal, insert this to the end of body
<div style="position: absolute;top: 0px;left: 0px;width: 100%;/* height: 50px; */">
   <div>
      <div class="ant-tooltip ant-tooltip-placement-bottom  ant-tooltip-hidden" style="left: -895px; top: -958.909px; transform-origin: 50% -4px;">
         <div class="ant-tooltip-content">
         <div class="ant-tooltip-arrow">
         <span class="ant-tooltip-arrow-content"></span>
      </div>
      <div class="ant-tooltip-inner" role="tooltip">PostHog is up-to-date</div>
   </div>
</div>
```

## Root Cause

For the first type of marker (annotations we already know exist), we render them on first mount of the graph since we know where these markers should be positioned. We see them being appended to the end of `<body>`'s children. 

![Screen Shot 2021-06-04 at 1 53 13 PM](https://user-images.githubusercontent.com/13460330/120861481-37371b00-c53c-11eb-84bb-ac79ec6ec33b.png)

For the second type of marker (modal to create an annotation), the modal's position depends on which date the user has clicked on to create an annotation.

Today's code dynamically renders an `AnnotationMarker` every time a user clicks the + sign and opens this marker. Because the marker's boundaries are defined by the full-size viewport (see context), the browser scrolls to the top of the page to display the largest bounding element (`<div style="position: absolute">`) containing this marker. 

![Screen Shot 2021-06-04 at 1 56 49 PM](https://user-images.githubusercontent.com/13460330/120861793-b7f61700-c53c-11eb-94c0-f6b6ca149d6e.png)

When the user clicks away or closes the modal, the `AnnotationMarker` is unmounted.

## Solution

~~Render a hidden create annotations marker on first mount of graph component so that it isn't rerendered everytime a user clicks to create a new annotation.~~

Using [`getPopupContainer`](https://ant.design/components/tooltip/#API) prop and defining modal in parent graph component so that root modal doesn't get rerendered on every hover action.

**Comment by mariusandra** [comment]
Closed with #4610
